components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/HTTPPostFunctionImpl.java

```diff
@@ -87,6 +87,18 @@ public void httpPost(String endpointURL, Object... params) {
                             "(Map<String, Object>) respectively.");
                 }
                 break;
+            case 4:
+                if (params[0] instanceof Map && params[1] instanceof Map && params[2] instanceof Map && params[3] instanceof Map) {
+                    payloadData = (Map<String, Object>) params[0];
+                    headers = validateHeaders((Map<String, ?>) params[1]);
+                    authConfig = getAuthConfigModel((Map<String, Object>) params[2]);
+                    eventHandlers = (Map<String, Object>) params[3];
+                }  else {
+                    throw new IllegalArgumentException("Invalid argument type. Expected payloadData " +
+                            "(Map<String, Object>), headers (Map<String, String>), and eventHandlers " +
+                            "(Map<String, Object>) respectively.");
```

Comment:
This message also doesn't seem to tally with the arguments the function expect.
The order is payload, headers, auth configuration and event handlers

Reply:
Resolved with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/157/commits/01b5e2f5f849d7a4e89d792e28845597544fca39